Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -107,6 +128,82 @@ public void Razor_ExcludedProject_NoErrorLog()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "0");
+            var temporaryProjectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, @"0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.Razor");
+            TestUtils.CreateEmptyFile(temporaryProjectSpecificOutDir, "Issues.FromMainBuild.json");
+            TestUtils.CreateEmptyFile(projectSpecificOutDir, "Issues.FromRazorBuild.json");
+            var razroIssuesPath = Path.Combine(razorSpecificOutDir, "Issues.FromRazorBuild.json");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <SonarTemporaryProjectSpecificOutDir>{temporaryProjectSpecificOutDir}</SonarTemporaryProjectSpecificOutDir>
+  <ProjectSpecificOutDir>{projectSpecificOutDir}</ProjectSpecificOutDir>
+  <RazorSonarErrorLog>{razroIssuesPath}</RazorSonarErrorLog>
+</PropertyGroup>
+
+<ItemGroup>
+  <RazorCompile Include='SomeRandomValue'>
+  </RazorCompile>
+</ItemGroup>
+";
+
+            var filePath = CreateProjectFile(null, projectSnippet, TargetConstants.SonarFinishRazorCodeAnalysis);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.SonarFinishRazorCodeAnalysis);
+
+            // Assert
+            var actualProjectInfo = ProjectInfoAssertions.AssertProjectInfoExists(root, filePath);
+            result.AssertTargetExecuted(TargetConstants.SonarFinishRazorCodeAnalysis);
+            actualProjectInfo.AnalysisSettings.Single(x => x.Id.Equals("sonar.cs.analyzer.projectOutPaths")).Value.Should().Be(razorSpecificOutDir);
+            Directory.Exists(temporaryProjectSpecificOutDir).Should().BeFalse();
+            File.Exists(Path.Combine(projectSpecificOutDir, "Issues.FromMainBuild.json")).Should().BeTrue();
+            File.Exists(Path.Combine(razorSpecificOutDir, "Issues.FromRazorBuild.json")).Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void Razor_RazorSpecificOutputAndProjectInfo_PreserveUserDefinedErrorLogValue()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "0");
+            var userDefinedOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "0_userDefined");
+            var temporaryProjectSpecificOutDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, @"0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.Razor");
+            var userDefinedErrorLog = TestUtils.CreateEmptyFile(userDefinedOutDir, "Issues.FromRazorBuild.json");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <RazorSonarErrorLog>{userDefinedErrorLog}</RazorSonarErrorLog>
```

Comment:
Users should not define our property, but the SDK one.
```suggestion
  <RazorCompilationErrorLog>{userDefinedErrorLog}</RazorCompilationErrorLog>
```

Reply:
Indded that is what the user should set.  In this case however the Target uses the `RazorSonarErrorLog` which is set  in the `SonarPrepareRazorCodeAnalysis` target from the `RazorCompilationErrorLog`. This UT runs only the `SonarFinishRazorCodeAnalysis` target so if I change it to `RazorCompilationErrorLog` it will fail.